src/main/java/org/squiddev/plethora/integration/roost/MethodsRoost.java

```diff
@@ -0,0 +1,72 @@
+package org.squiddev.plethora.integration.roost;
+
+import com.setycz.chickens.ChickensMod;
+import com.setycz.chickens.registry.ChickensRegistry;
+import com.setycz.chickens.registry.ChickensRegistryItem;
+import com.timwoodcreates.roost.Roost;
+import com.timwoodcreates.roost.data.DataChicken;
+import com.timwoodcreates.roost.data.DataChickenModded;
+import com.timwoodcreates.roost.data.DataChickenVanilla;
+import com.timwoodcreates.roost.tileentity.TileEntityBreeder;
+import net.minecraftforge.fml.common.Loader;
+import org.squiddev.plethora.api.method.IContext;
+import org.squiddev.plethora.api.method.wrapper.PlethoraMethod;
+import org.squiddev.plethora.utils.LuaList;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.Map;
+
+public class MethodsRoost {
+
+	@PlethoraMethod(
+			modId = Roost.MODID,
+			doc = "-- Get a list of chickens species"
+	)
+	public static Map<Integer, Object> getSpeciesList(@Nonnull IContext<TileEntityBreeder> context) {
+		if (Loader.isModLoaded(ChickensMod.MODID)) {
```

Comment:
OK, so what you could _technically_ do here is register `TileEntityBreeder` as a analyser module, which is what we do for noteblocks:

https://github.com/SquidDev-CC/plethora/blob/ad8bbffeb134dda08c7bc6e5d3e7148848ff9b6d/src/main/java/org/squiddev/plethora/integration/vanilla/IntegrationVanilla.java#L74

Oh goodness, I have created a monster.

Anyway, your call - there may be stuff you want to add to this method that the chicken analyser doesn't have.

Reply:
There's a little bit of difference between the two, as Roost also specifies information for the vanilla chicken.  I've just added comments reminding anyone who works on the code to check the other class as well.